File: Externals/4pdaCommon/src/org/softeg/slartus/forpdacommon/AppHttpStatus.java
package org.softeg.slartus.forpdacommon;

/**
 * Created by IntelliJ IDEA.
 * User: Admin
 * Date: 27.11.11
 * Time: 22:45
 * To change this template use File | Settings | File Templates.
 */
public class AppHttpStatus {
    public static String getReasonPhrase(int code, String defaultPhrase) {
        switch (code) {
            case 404:
                return "Страница не найдена";
            case 500:
                return "Внутренняя ошибка сервера";
            case 501:
                return "Не реализовано";
            case 502:
                return "Плохой шлюз";
            case 503:
                return "Сервис недоступен";
            case 504:
                return "Шлюз не отвечает";
            case 505:
                return "Версия HTTP не поддерживается";
            case 506:
                return "Вариант тоже согласован";
            case 507:
                return "Переполнение хранилища";
            case 509:
                return "Исчерпана пропускная ширина канала";
            case 510:
                return "Не расширено";
        }
        return defaultPhrase;
    }
}

File: http/src/main/java/ru/slartus/http/AppHttpStatus.kt
package ru.slartus.http

/*
 * Created by slinkin on 01.07.13.
 */
object AppHttpStatus {
    fun getReasonPhrase(code: Int, defaultPhrase: String): String {
        when (code) {
            500 -> return "Внутренняя ошибка сервера"
            501 -> return "Не реализовано"
            502 -> return "Плохой шлюз"
            503 -> return "Сервис недоступен"
            504 -> return "Шлюз не отвечает"
            505 -> return "Версия HTTP не поддерживается"
            506 -> return "Вариант тоже согласован"
            507 -> return "Переполнение хранилища"
            509 -> return "Исчерпана пропускная ширина канала"
            510 -> return "Не расширено"
        }
        return defaultPhrase
    }
}
